private/protocol/query/unmarshal_error.go

```diff
@@ -36,5 +45,22 @@ func UnmarshalError(r *request.Request) {
 			r.HTTPResponse.StatusCode,
 			reqID,
 		)
+		return
+	}
+
+	// Check for unhandled error
+	servUnavailResp := xmlServiceUnavailableResponse{}
+	unavailErr := xml.Unmarshal(bodyBytes, &servUnavailResp)
```

Comment:
Interesting didn't know you could do that with XML tags, `ServiceUnavailableException` isn't an attribute, but a different xml tag than `ErrorResponse`.

I'm  not sure if xml can do that. I didn't see any mention of it in the godoc, and this play example hits an error, http://play.golang.org/p/NCECBedwc5